### ISSUE TITLE
Shell evasion by $IFS / ${IFS}

### ISSUE THREAD

**Issue by fzipi** [crstest01/owasp-modsecurity-crs#669]
_Issue originally created by user lifeforms on date 2016-11-25 19:52:04.
            Link to original issue: https://github.com/SpiderLabs/owasp-modsecurity-crs/issues/664._

In bash, `$IFS` can be used as a replacement for the space character. This trick is actively used.

Example:

```
GET /language/Swedish${IFS}&&echo${IFS}610cker>qt&&tar${IFS}/string.js HTTP/1.0
Accept: */*
User-Agent: Wget(linux)
```

I propose to add `$IFS` and some [interesting other variables](http://www.tldp.org/LDP/abs/html/internalvariables.html) to `unix-shell.data` and also add REQUEST_URI to scanned variables.

The expression `${IFS}` is already caught by rule 932130, but to catch this example, we should also add REQUEST_URI to scanned variables.

**Comment by fzipi** [comment]
_User dune73 commented on date 2016-11-26 12:12:13:_

Good catch. I was not aware of this possibility of `$IFS`.

What other variables do you have in mind?

I am a bit reluctant to extend an existing rule with REQUEST_URI for a point release. So are you aiming for 3.0.1 or would you wait with that for 3.1.0

**Comment by fzipi** [comment]
_User csanders-git commented on date 2016-11-26 16:19:53:_

interesting note. I'm glad you shared it.

**Comment by fzipi** [comment]
_User lifeforms commented on date 2019-03-12 19:23:59:_

Implemented in #1325
